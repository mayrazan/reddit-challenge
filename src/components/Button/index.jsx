import styled, { css } from 'styled-components';
import { TextStyleVariants } from '../Text';

const Button = styled.button`
  color: ${({ theme, color }) => theme.text[color]};
  border-radius: ${({ theme }) => theme.borderRadius};
  transition: opacity ${({ theme }) => theme.transition};
  background-color: ${({ theme }) => theme.background.gray};
  ${TextStyleVariants('subtitle')}
  width: 202px;
  height: 48px;
  border: 0;
  cursor: pointer;
  opacity: 1;

  &:hover {
    opacity: 0.5;
  }
  &:disabled {
    cursor: not-allowed;
    opacity: 0.2;
  }
  ${({ fullWidth }) => fullWidth
    && css`
      width: 100%;
      background-color: ${({ theme }) => theme.background.shadesPurple};
    `};

  ${({ isActive }) => isActive
    && css`
      background-color: ${({ theme }) => theme.background.shadesPurple};
    `};
`;

export default Button;
