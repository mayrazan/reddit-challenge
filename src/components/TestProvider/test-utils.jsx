import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import theme from '../../theme';
import GlobalStyle from '../../theme/GlobalStyle';
import { lightColors } from '../../theme/colors';

const AllTheProviders = ({ children, ...props }) => (
  <ThemeProvider theme={{ ...theme, ...lightColors }} {...props}>
    <GlobalStyle />
    {children}
  </ThemeProvider>
);
AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
