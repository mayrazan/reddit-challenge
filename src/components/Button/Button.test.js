import React from 'react';
import { fireEvent } from '@testing-library/dom';
import { render, screen } from '../TestProvider/test-utils';

import Button from '.';

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('<Button />', () => {
  test('renders component', () => {
    render(
      <Button color="shadesPurple" fullWidth isActive>
        New
      </Button>,
    );
    const button = screen.getByText(/new/i);
    expect(button).toMatchSnapshot();
    expect(button).toBeInTheDocument();
  });

  test('calls "onClick" prop on button click', () => {
    render(<Button onClick={() => mockHistoryPush('/new')}>New</Button>);

    fireEvent.click(screen.getByRole('button'));
    expect(mockHistoryPush).toHaveBeenCalledWith('/new');
  });
});
