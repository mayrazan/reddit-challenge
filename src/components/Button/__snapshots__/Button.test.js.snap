// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders component 1`] = `
.c0 {
  border-radius: 8px;
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
  background-color: #A7B0BE;
  font-size: 16px;
  font-weight: normal;
  line-height: 20px;
  font-weight: 600;
  width: 202px;
  height: 48px;
  border: 0;
  cursor: pointer;
  opacity: 1;
  width: 100%;
  background-color: #6324C6;
  background-color: #6324C6;
}

.c0:hover {
  opacity: 0.5;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.2;
}

@media screen and (min-width:768px) {
  .c0 {
    font-size: 20px;
    font-weight: 600;
    line-height: 25px;
  }
}

<button
  class="c0"
  color="shadesPurple"
>
  New
</button>
`;
